From 8594d4210437f6be4e295e4ed39d9e6807c1e9ef Mon Sep 17 00:00:00 2001
From: twaters <travis.waters@sdl.usu.edu>
Date: Mon, 21 Dec 2015 14:41:42 -0700
Subject: [PATCH 1/2] Update for win32

---
 src/c/include/winconfig.h | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/c/include/winconfig.h b/src/c/include/winconfig.h
index cc92396..d3e9229 100644
--- a/src/c/include/winconfig.h
+++ b/src/c/include/winconfig.h
@@ -157,13 +157,20 @@
 #endif
 
 #define NTDDI_VERSION _WIN32_WINNT_NT4
+
+#ifndef _WIN32_WINNT
 #define _WIN32_WINNT _WIN32_WINNT_NT4
+#endif
 
+#ifndef _CRT_SECURE_NO_WARNINGS
 #define _CRT_SECURE_NO_WARNINGS
+#endif
+#ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN
+#endif
 #include <Windows.h>
 #include <Winsock2.h>
-#include <winstdint.h>
+#include <stdint.h>
 #include <process.h>
 #include <ws2tcpip.h>
 #undef AF_INET6
-- 
1.9.5.msysgit.0


From 30a6f4c79f739f8f0fdf4a91f10e39780469b34f Mon Sep 17 00:00:00 2001
From: twaters <travis.waters@sdl.usu.edu>
Date: Mon, 21 Dec 2015 14:52:13 -0700
Subject: [PATCH 2/2] More winconfig updates

---
 src/c/include/winconfig.h | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/src/c/include/winconfig.h b/src/c/include/winconfig.h
index d3e9229..5ee71e9 100644
--- a/src/c/include/winconfig.h
+++ b/src/c/include/winconfig.h
@@ -1,3 +1,5 @@
+#pragma once
+
 /* Define to 1 if you have the <arpa/inet.h> header file. */
 #undef HAVE_ARPA_INET_H
 
@@ -17,7 +19,7 @@
 #undef HAVE_GETCWD
 
 /* Define to 1 if you have the `gethostbyname' function. */
-#undef HAVE_GETHOSTBYNAME
+#define HAVE_GETHOSTBYNAME 1
 
 /* Define to 1 if you have the `gethostname' function. */
 #define HAVE_GETHOSTNAME 1
@@ -41,10 +43,10 @@
 #undef HAVE_MEMMOVE
 
 /* Define to 1 if you have the <memory.h> header file. */
-#undef HAVE_MEMORY_H
+#define HAVE_MEMORY_H 1
 
 /* Define to 1 if you have the `memset' function. */
-#undef HAVE_MEMSET
+#define HAVE_MEMSET 1
 
 /* Define to 1 if you have the <netdb.h> header file. */
 #undef HAVE_NETDB_H
@@ -59,7 +61,7 @@
 #undef HAVE_SOCKET
 
 /* Define to 1 if you have the <stdint.h> header file. */
-#undef HAVE_STDINT_H
+#define HAVE_STDINT_H 1
 
 /* Define to 1 if you have the <stdlib.h> header file. */
 #define HAVE_STDLIB_H 1
@@ -74,13 +76,13 @@
 #define HAVE_STRERROR 1
 
 /* Define to 1 if you have the <strings.h> header file. */
-#undef HAVE_STRINGS_H
+#define HAVE_STRINGS_H 1
 
 /* Define to 1 if you have the <string.h> header file. */
-#undef HAVE_STRING_H
+#define HAVE_STRING_H 1
 
 /* Define to 1 if you have the `strtol' function. */
-#undef HAVE_STRTOL
+#define HAVE_STRTOL 1
 
 /* Define to 1 if you have the <sys/socket.h> header file. */
 #undef HAVE_SYS_SOCKET_H
@@ -156,7 +158,9 @@
 #define _WIN32_WINNT_NT4 0x0400
 #endif
 
+#ifndef NTDDI_VERSION
 #define NTDDI_VERSION _WIN32_WINNT_NT4
+#endif
 
 #ifndef _WIN32_WINNT
 #define _WIN32_WINNT _WIN32_WINNT_NT4
-- 
1.9.5.msysgit.0

