diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..9597f63
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,41 @@
+cmake_minimum_required(VERSION 3.0)
+project(novas)
+set(CMAKE_DEBUG_POSTFIX "d")
+##############################
+# A list of files that can be used in all plugins
+set(${PROJECT_NAME}_srcs
+  eph_manager.c
+  eph_manager.h
+  novas.c
+  novas.h
+  novascon.c
+  novascon.h
+  nutation.c
+  nutation.h
+  readeph0.c
+  solarsystem.h
+  solsys3.c
+  )
+############################## 
+# Create an object library to include in the plugin shared objects.
+add_library(${PROJECT_NAME} STATIC ${${PROJECT_NAME}_srcs})
+############################## 
+install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}-targets
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+)
+set(header_files
+  eph_manager.h
+  novas.h
+  novascon.h
+  nutation.h
+  solarsystem.h
+)
+install(FILES ${header_files} DESTINATION include/novas)
+# Compiler options
+if(UNIX)
+  # Creates position independent code in GCC making it suitable for
+  # inclusion in a library or shared object.
+  set_property(TARGET ${PROJECT_NAME} PROPERTY COMPILE_FLAGS "-fPIC")
+endif()
