diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 000000000..0486822e9
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,2 @@
+cmake_minimum_required(VERSION 3.5)
+add_subdirectory(zookeeper-client/zookeeper-client-c)
\ No newline at end of file
diff --git a/zookeeper-client/zookeeper-client-c/CMakeLists.txt b/zookeeper-client/zookeeper-client-c/CMakeLists.txt
index 12773fc92..c0e5012c1 100644
--- a/zookeeper-client/zookeeper-client-c/CMakeLists.txt
+++ b/zookeeper-client/zookeeper-client-c/CMakeLists.txt
@@ -181,6 +181,9 @@ target_link_libraries(zookeeper PUBLIC
   hashtable
   $<$<PLATFORM_ID:Linux>:rt> # clock_gettime
   $<$<PLATFORM_ID:Windows>:ws2_32>) # Winsock 2.0
+install(TARGETS zookeeper DESTINATION lib)
+install(DIRECTORY include/ DESTINATION include/zookeeper CONFIGURATIONS Release)
+install(FILES generated/zookeeper.jute.h DESTINATION include/zookeeper CONFIGURATIONS Release)
 
 option(WITH_OPENSSL "turn ON/OFF SSL support, or define openssl library location (default: ON)" ON)
 message("-- using WITH_OPENSSL=${WITH_OPENSSL}")
